swap_meet/vendor.py

```diff
@@ -1,2 +1,91 @@
+from .item import Item
+
 class Vendor:
-    pass
+    def __init__(self, inventory=[]):
+        self.inventory = inventory if inventory else []
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+    def get_by_category(self, category):
+        category_items = []
+
+        for item in self.inventory:
+            if item.category ==  category:
+                category_items.append(item)
+        return category_items
+
+    def swap_items(self, vendor, my_item, their_item):
+        self.vendor = vendor
+        self.my_item = my_item
+        self.their_item = their_item
+
+        if my_item in self.inventory and their_item in vendor.inventory:
+            vendor.add(my_item)
+            self.add(their_item)
+            self.remove(my_item)
+            vendor.remove(their_item)
+            return True
+        return False
+
+    def swap_first_item(self, vendor):
+        if len(self.inventory) > 0 and len(vendor.inventory) > 0:
+            self.swap_items(vendor, self.inventory[0], vendor.inventory[0])
+            return True
+        return False
+
+    def get_best_by_category(self, category):
+        best = None
+
+        for item in self.get_by_category(category):
+            if best == None:
+                best = item
+            elif item.condition > best.condition:
+                best = item
+        return best
+
+        # ****Other option, not sure which one is best
+        # list_items = self.get_by_category(category)
+
+        # if len(list_items) > 0:
+        #     best = list_items[0]
+        #     for item in list_items:
+        #         if item.condition > best.condition:
+        #             best = item
+        #     return best
+        # return None
```

Comment:
Even though both are good implementations. I like the commented code because of the len check!